packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -97,12 +167,35 @@ export async function attachNodeBlockDeadlineToEvent(
     return event
   }
 
+  const firstExecutableBlockFromCache = getFirstExecutableBlockFromCache({
+    event,
+    l2ChainID
+  })
+
+  if (firstExecutableBlockFromCache) {
+    return {
+      ...event,
+      nodeBlockDeadline: firstExecutableBlockFromCache
+    }
+  }
+
   const messageReader = L2ToL1MessageReader.fromEvent(l1Provider, event)
 
   try {
     const firstExecutableBlock = await messageReader.getFirstExecutableBlock(
       l2Provider
     )
+    const firstExecutableBlockCacheKey = getMessagesCacheKey({
+      event,
+      l2ChainId: l2ChainID
+    })
+
+    if (firstExecutableBlockCacheKey && firstExecutableBlock) {
+      saveFirstExecutableBlockToCache(
+        firstExecutableBlockCacheKey,
+        firstExecutableBlock.toNumber()
```

Comment:
This could overflow so better to cache the value of `.toString()` and later parse into a `BigNumber`

Reply:
done 'toString', we discussed 'BigNumber' on priv and won't change that